doc/docusaurus/docs/using-plutus-tx/running-debugging.md

```diff
@@ -0,0 +1,328 @@
+if your are using Plutus Tx as source, you can follow the instructions on [how to extract compiled code].
+
+Although the CLI tool cannot help with compiling high-level code (e.g. Plutus Tx), it can aid with compiling and converting
+between different formats and intermediate representations of plutus code.
+
+In the following example we check a program written in the PIR (Plutus Intermediate Representation) language:
+
+``` shell
+$ echo "(program 1.1.0 (lam x (con integer) (lam y (con integer) x)))" > const.pir
+$ plutus const.pir
+Program passed all static checks. If you want the output result, use -o FILE or --stdout.
+```
+
+Or as a single step:
+
+``` shell
+$ echo "(program 1.1.0 (lam x (con integer) (lam y (con integer) x)))" | plutus -x pir --stdin
+Program passed all static checks. If you want the output result, use -o FILE or --stdout.
+```
+
+In the command above, the `-x pir` option is necessary for letting the CLI tool know
+the format/language of the supplied input. If the `-x` option is not passed, the CLI tool will try to guess the input's format/language
+by looking at its filename extension `.SUFFIX` --- and will default to `plc` for `--stdin` or no suffix.
+What the `-x SUFFIX` option does is instructing the CLI tool to treat subsequent filenames
+as if they had a `.SUFFIX` file extension.
+The following table lists the recognized suffixes and corresponding `-x` values:
+
+|Filename Suffix|-x Value|Format Type|Description|
+|---|---|---|---|
+
+
+FIXME table
+
+In case the input format is more complex (contains a non-default variable-naming scheme or annotations),
+the `-n NAMING` and `-a ANNOTATION` options can be used respectively to override the defaults.
+
+FIXME naming and annotation table
+
+Note that the `-x`,`-n`,`-a` options are *positional*: if
+there are multiple input filenames, multiple calls to these options can be used
+to override previous occurrences so that you can combine code of different formats. For example:
+
+``` shell
+$ plutus -x pir FILE_PIR FILE_ALSO_PIR -x tplc FILE_NOW_TPLC -a SrcSpans FILE_TPLC_BUT_WITH_ANNOTATIONS
+```
+
+Positionining the input files one after the other (juxtaposing) as such, acts
+like Haskell's function application:
+it compiles individually each plutus snippet to a common *output target* and combines left-to-right
+their compilation result with plutus' function application. Both inputs and final compiled output are (type)checked.
+
+``` shell
+# plutus FUNC_FILE ARG1_FILE ARG2_FILE ...
+$ echo "(program 1.1.0 (lam x (lam y [(builtin subtractInteger) x y])))" > func.uplc
+$ echo "(program 1.1.0 (con integer 2))" > arg2.uplc
+$ echo "(program 1.1.0 (con integer 1))" | plutus func.uplc --stdin arg2.uplc
+```
+
+To override the output's target format the `-x`,`-n`,`-a` options are again used;
+note that the *last occurrence* of `-x`,`-n`,`-a` will be the one that dictates the output format.
+
+``` shell
+# Checks and compiles PIR to PLC (the default)
```

Comment:
You mean UPLC? (PLC is confusing imo, you don't know whether it's the typed or the untyped version!)

Reply:
Y wherever u see plc I meant uplc :man_facepalming: